src/main/java/game/ScoreCounter.java

```diff
@@ -0,0 +1,155 @@
+import java.nio.charset.StandardCharsets;
+
+import display.DisplayText;
+
+/**
+ * Class that maintains the score.
+ * @author Esmee
+ *
+ */
+public class ScoreCounter {
+	/**
+	 * The amount of points needed to gain a life.
+	 */
+	private static final int LIFE_SCORE = 10000;
+
+	private long score;
+	private long highscore;
+	
+	/**
+	 * Constructor for score counter.
+	 */
+	public ScoreCounter() {
+		highscore = readHighscore();
+	}
+	
+	/**
+	 * reads the highscore from file in resources folder.
+	 *
+	 * @return the highscore
+	 */
+	private long readHighscore() {
+		long currentHighscore = 0;
+		final String filePath = "src/main/resources/highscore.txt";
+		try (BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(filePath),
+				StandardCharsets.UTF_8))) {
+			String sCurrentLine;
+			while ((sCurrentLine = br.readLine()) != null) {
+				currentHighscore = Long.parseLong(sCurrentLine);
+			}
+		} catch (IOException e) {
+			Logger.getInstance().log("unable to read highscore from file", e);
+		}
+		return currentHighscore;
+	}
+	
+	/**
+	 * writes the highscore to file in resources folder.
+	 */
+	private void writeHighscore() {
+		final String content = String.valueOf(highscore);
+		final File file = new File("src/main/resources/highscore.txt");
+		try (FileOutputStream fos = new FileOutputStream(file.getAbsoluteFile())) {
+			fos.write(content.getBytes(StandardCharsets.UTF_8));
+			fos.flush();
+			fos.close();
+		} catch (IOException e) {
+			Logger.getInstance().log("unable to write highscore to file", e);
+		}
+	}
+	
+	/**
+	 * Describes what should happen at the start of the game.
+	 */
+	protected final void startGame() {
+		if (this.score > highscore) {
+			highscore = this.score;
+			writeHighscore();
+		}		
+		score = 0;
+	}
+	
+	/**
+	 * Display score on screen.
+	 */
+	public final void displayScore() {
+		DisplayText.score(score);
+		DisplayText.highscore(highscore);
+	}
+	
+	/**
+	 * @return the high score
+	 */
+	public final long getHighscore() {
+		return highscore;
+	}
+	
+	/**
+	 * Score getter.
+	 *
+	 * @return score
+	 */
+	public final long getScore() {
+		return score;
+	}
+	
+	/**
+	 * @return true if the score is better than the highscore
+	 */
+	public final boolean isHighscore() {
+		return score > highscore;
+	}
+	
+	/**
+	 * @return true if the score is not better than the high score
+	 */
+	public final boolean isNotHighscore() {
+		return !isHighscore();
+	}
+	
+	/**
+	 * Update the highscore.
+	 */
+	protected final void updateHighscore() {
+		highscore = score;
+		writeHighscore();
+	}
+	
+	/**
+	 * Add the amount of points to the current score.
+	 * @param points amount of points to be added
+	 */
+	public final void addScore(final long points) {
+		score += points;
+	}
+	
+	/**
+	 * @param points amount of points that the player gains
+	 * @return true when the player can gain a life
+	 */
+	public final boolean canGainLife(final int points) {
```

Comment:
maak alsjeblieft in de javadoc iets duidelijker dat deze methode checked of the player genoeg punten heeft om een extra leven te krijgen